recirq/otoc/parallel_xeb.py

```diff
@@ -13,175 +13,219 @@
 # limitations under the License.
 
 """Functions for performing parallel cross entropy benchmarking."""
-
+from dataclasses import dataclass
 from typing import Sequence, List, Set, Tuple, Dict, Union, Optional
 
 import cirq
 import numpy as np
 import pybobyqa
+from cirq.experiments.cross_entropy_benchmarking import _default_interaction_sequence
 from matplotlib import pyplot as plt
 
-from recirq.otoc.utils import (bits_to_probabilities, angles_to_fsim,
-                               pauli_error_fit, generic_fsim_gate)
+from recirq.otoc.utils import (
+    bits_to_probabilities,
+    angles_to_fsim,
+    pauli_error_fit,
+    generic_fsim_gate,
+    cz_to_sqrt_iswap,
+)
 
 _rot_ops = [
-    cirq.X ** 0.5, cirq.PhasedXPowGate(phase_exponent=0.25, exponent=0.5),
-    cirq.Y ** 0.5, cirq.PhasedXPowGate(phase_exponent=0.75, exponent=0.5),
-    cirq.X ** -0.5, cirq.PhasedXPowGate(phase_exponent=-0.75, exponent=0.5),
-    cirq.Y ** -0.5, cirq.PhasedXPowGate(phase_exponent=-0.25, exponent=0.5),
+    cirq.X ** 0.5,
+    cirq.PhasedXPowGate(phase_exponent=0.25, exponent=0.5),
+    cirq.Y ** 0.5,
+    cirq.PhasedXPowGate(phase_exponent=0.75, exponent=0.5),
+    cirq.X ** -0.5,
+    cirq.PhasedXPowGate(phase_exponent=-0.75, exponent=0.5),
+    cirq.Y ** -0.5,
+    cirq.PhasedXPowGate(phase_exponent=-0.25, exponent=0.5),
 ]
 _rot_mats = [cirq.unitary(r) for r in _rot_ops]
-_fsim_angle_labels = ['theta', 'delta_plus', 'delta_minus_off_diag',
-                      'delta_minus_diag', 'phi']
+_fsim_angle_labels = [
+    "theta",
+    "delta_plus",
+    "delta_minus_off_diag",
+    "delta_minus_diag",
+    "phi",
+]
 
 
-def default_interaction_sequence(
-        qubits: Sequence[cirq.GridQubit]
-) -> List[Set[Tuple[cirq.GridQubit, cirq.GridQubit]]]:
-    """Group qubits on a grid into sets of pairs.
+@dataclass
+class XEBData:
+    """Class for storing the cycle-dependent fidelities and purities of an XEB experiment."""
```

Comment:
Why are these all tuples of two ndarrays? Can you add a little note to the docstring?

Reply:
They are just the X (cycle) and Y (XEB fidelity or speckle purity) values of the decay curves and their fits. I added an explanation here.